modules/migrate.py

```diff
@@ -0,0 +1,485 @@
+
+        # For all object tags, clean up params and including config file. Use
+        # the old relative path because BeautifulSoup parse tree was created
+        # before the regex replace above and doesn't know about it.
+        for object in soup.find_all('object',
+                attrs={'data': widgetViewRelativePath}):
+            # Find all param tags for that widget
+            for param in object.find_all('param'):
+                # Param value could be anything, but it might be a relative
+                # path. Treat it as such creating an absolute path and see if it
+                # exists. If it does, assume it is a relative path to something
+                # in the project and update it to be relative to the new modular
+                # widget location.
+                val = param['value']
+                path = os.path.normpath(os.path.join(widgetAbsolutePath, val))
+
+                # NOTE(andy): For this test to work we have to check out the
+                # whole book. Sad times. But now we do.
+                if os.path.exists(path):
+                    newPath = os.path.relpath(path, modularWidgetAbsolutePath)
+
+                    # Target specifically the parameter with a relative path
+                    # inside the <object> tag we are migrating. If we didn't
+                    # include the <object> tag we could accidentally change
+                    # a relative path for a widget that wasn't migrating that
+                    # referenced the same asset / content.
+                    toReplace = (r'(<object.*?data="' + relativeModuleViewPath +
+                        '".*?<param.*?value=")' + val + '(".*?)(?=</object>)')
+                    lines = re.sub(toReplace,
+                                   r'\g<1>' + newPath + r'\g<2>',
+                                   lines,
+                                   flags=re.IGNORECASE|re.DOTALL)
+
+                    # If the param is for the widget config file, also clean up
+                    # that config file.
+                    if param['name'] == 'configFile':
+                        _updateConfigFile(path, widgetAbsolutePath,
+                                           modularWidgetAbsolutePath)
+
+        # Write out all the changes.
+        log.info('Writing updated HTML file: %s', filePath)
+        with open(filePath, 'wb') as htmlFile:
+           htmlFile.write(lines)
+
+def _updateConfigFile(filePath, widgetPath, modularWidgetPath):
+    """Updates all values nested in the JSON object that are paths relative to
+    the widgetPath to instead be relative to modularWidgetPath.
+
+        Args:
+            filePath - String absolute path to the JSON config file.
+            widgetPath - String absolute path to the non-modular widget.
+            modularWidthPath - String absolute path to the modular widget.
+    """
+    log.debug('Reading JSON config file: %s', filePath)
+    with open(filePath, 'rb') as configFile:
```

Comment:
`_updateHTMLFile` is wrapped within a try/except, so if this throws an IOError, things should be okay. Do we, however, want to try catching bad config file paths here so that the rest of the changes still get written?